protocol/app/app.go

```diff
@@ -945,7 +990,7 @@ func New(
 	// must be passed by reference here.
 	app.ModuleManager = module.NewManager(
 		genutil.NewAppModule(
-			app.AccountKeeper, app.StakingKeeper, app.BaseApp.DeliverTxShouldLock,
```

Comment:
To confirm, why are we removing this?

Reply:
Since the lock we added to Cosmos SDK for `CheckTx` parallelism also needed to be added to `DeliverTx` but there were use cases where `DeliverTx` is re-entrant and hence already held the lock.

Now that Cosmos SDK uses `FinalizeBlock`, the lock that was elevated to `FinalizeBlock` and now the it doesn't matter that `DeliverTx` is re-entrant since it no longer acquires the lock.